cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/PluginExclusionTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.internal.app.services.http.handlers;
+
+import co.cask.cdap.api.annotation.Requirements;
+import co.cask.cdap.api.artifact.ArtifactRange;
+import co.cask.cdap.api.artifact.ArtifactVersion;
+import co.cask.cdap.api.plugin.PluginClass;
+import co.cask.cdap.app.program.ManifestFields;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.id.Id;
+import co.cask.cdap.internal.app.runtime.artifact.app.inspection.InspectionApp;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.NamespaceId;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+
+import java.util.Set;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+
+/**
+ * Tests that plugins whose requirements are marked as excluded through cdap-site.xml are excluded and is not visible
+ */
+public class PluginExclusionTest extends ArtifactHttpHandlerTestBase {
+  @ClassRule
+  public static final ExternalResource RESOURCE = new ExternalResource() {
+    @Override
+    protected void before() {
+      // Set the system property to the excluded requirement value which will be set in cdap-site through the test base
+      System.setProperty(Constants.REQUIREMENTS_BLACKLIST, Requirements.TEPHRA_TX);
+    }
+
+    @Override
+    protected void after() {
+      // clear the system property
+      System.clearProperty(Constants.REQUIREMENTS_BLACKLIST);
```

Comment:
You should remember the old property in the `before()` and restore it here (or if the property did not exists in `before()`, then clear it here).

Reply:
Thanks. Done.